scripts/vale/vale.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const chalk = require( 'chalk' );
+const upath = require( 'upath' );
+const minimist = require( 'minimist' );
+const { globSync } = require( 'glob' );
+const { spawn } = require( 'child_process' );
+
+const CKEDITOR5_ROOT = upath.join( __dirname, '..', '..' );
+
+const minimistOptions = {
+	string: [
+		'directory',
+		'files'
+	],
+	alias: {
+		d: 'directory',
+		f: 'files'
+	},
+	default: {
+		directory: '',
+		files: ''
+	}
+};
+
+const spawnOptions = {
+	cwd: CKEDITOR5_ROOT,
+	shell: true
+};
+
+const globOptions = {
+	cwd: CKEDITOR5_ROOT,
+	ignore: '**/node_modules/**'
+};
+
+const itemsToCheckInsidePackages = [
+	'README.md',
+	'docs'
+];
+
+const defaultPatterns = [
+	'CHANGELOG.md',
+	'CONTRIBUTING.md',
+	'LICENSE.md',
+	'README.md',
+	'SECURITY.md',
+	'docs',
+	'external/ckeditor5-commercial/docs',
+	...getPathsFromPackages( 'packages' ),
+	...getPathsFromPackages( 'external/ckeditor5-commercial/packages' )
+];
+
+main();
+
+function main() {
+	const args = minimist( process.argv.slice( 2 ), minimistOptions );
+	const patterns = args.directory ?
+		[ `**/${ args.directory }/**/*.md` ] :
+		args.files.split( ',' ).filter( Boolean );
+	const valeArgs = [ 'run', 'docs:vale' ];
+
+	if ( patterns.length ) {
+		const files = patterns.flatMap( pattern => globSync( pattern, globOptions ) );
+
+		if ( !files.length ) {
+			console.log( chalk.red( '\nProvided pattern did not match any files.\n' ) );
+
+			process.exit( 1 );
+		}
+
+		console.log( chalk.blue( `\nMatched ${ files.length } files.` ) );
+
+		valeArgs.push( ...files );
+	} else {
+		valeArgs.push( ...defaultPatterns );
+	}
+
+	console.log( chalk.blue( '\nExecuting vale...\n' ) );
+
+	const vale = spawn( 'yarn', valeArgs, spawnOptions );
```

Comment:
This part is tricky. To correctly pass glob pattern to the script, you need to escape them before the shell processes them.

**Example**: Executing `yarn docs:lint -f *.md` causes that shell automatically expands the `*.md` pattern into `CHANGELOG.md CONTRIBUTING.md LICENSE.md README.md SECURITY.md` files separated by a space. Then, this list of files is passed to the script in the following way: `node scripts/vale/vale.js -f CHANGELOG.md CONTRIBUTING.md LICENSE.md README.md SECURITY.md`. So, this script get `CHANGELOG.md` as the glob pattern, so Vale gets only the `CHANGELOG.md` file to process.

<img src="https://github.com/ckeditor/ckeditor5/assets/56868128/65660fc6-ed77-40b3-8696-052908e2225b" width="700px">

To make it work as we expect, glob pattern must be escaped.

**Example**: Executing `yarn docs:lint -f "*.md"` is not expanded by the shell, but the glob pattern is passed "as is" to this script.

<img src="https://github.com/ckeditor/ckeditor5/assets/56868128/f9be2ca4-9446-4723-99ce-e10f7c8664fc" width="300px">

The problem is when the `"**"` pattern is correctly escaped: the `ENAMETOOLONG` error is thrown:

<img src="https://github.com/ckeditor/ckeditor5/assets/56868128/ea0ccac4-b6d0-4f92-833f-c5d16d469614" width="700px">

To fix that, probably `spawn()` has to be wrapped inside `try ... catch`.